File: rb/features.rb
# Letscount SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module LetscountFeatures
  def self.make_feature(name)
    case name
    when "base"
      LetscountBaseFeature.new
    when "test"
      LetscountTestFeature.new
    else
      LetscountBaseFeature.new
    end
  end
end

File: py/features.py
# Letscount SDK feature factory

from feature.base_feature import LetscountBaseFeature
from feature.test_feature import LetscountTestFeature


def _make_feature(name):
    features = {
        "base": lambda: LetscountBaseFeature(),
        "test": lambda: LetscountTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
